### ISSUE TITLE
node-fast could track basic request metrics

### ISSUE THREAD

**Issue by KodyKantor** [joyent/node-fast#9]
It would be useful for consumers of node-fast to be able to automatically track metrics like request counts and request latency. [node-artedi](https://github.com/joyent/node-artedi) was built to allow node-based utilities to pass metric collectors into dependent libraries.

node-fast has access to generic request metadata (things like origin IP, request start/end times, RPC name), but applications may want to add additional arbitrary metadata to the metrics being tracked.

Using node-artedi some arbitrary metadata can be inherited automatically when collectors are created (by the parent utility). Other metadata will need to be passed in to node-fast from the parent utility when requests complete. Examples of the former in Moray terms are datacenter, zonename, PID. One example of the latter in Moray terms is the name of the bucket being operated upon. This will have to be accounted for in the implementation.

I think we can go ahead and allow node-fast to take an artedi-style collector as an optional argument, and have node-fast track a couple basic metrics to begin with (request counts and latency). Other metrics could be added in the future.

**Comment by KodyKantor** [comment]
Testing notes:
* I ran 'make prepush' on ever commit, which executes the in-repo testing suite
* I built and deployed a new Moray image using this version of fast (pointing Moray's `package.json` to the dev branch on GitHub)
* Ran the moray-test-suite against both standalone, and remote Moray instances using this node-fast version
* Used a Prometheus server to scrape metrics from Moray and create simple graphs of the metrics exposed from fast

Everything seems to work. Let me know if I should do any additional testing in dev or staging.

**Comment by davepacheco** [comment]
Nice!  The only other thing I would suggest is a basic memory leak check.  You could do that by running `fastbench` for a while or using Moray's `stress-client.js` test.

**Comment by KodyKantor** [comment]
Sounds good. I ran `fastbench` against this version of node-fast for six minutes with 100 concurrent requests at a time. The RSS for `fastserve` hovered around 55-62MB, never exceeding 62MB on my machine.

I also ran the `stress-client.js` test against a standalone Moray (with these changes) for five minutes. The RSS for the two Moray processes hovered around 61-72MB, never exceeding 72MB on my machine.
